File: bitget-node-sdk-api/src/lib/v3/UtaBrokerApi.ts
import {BaseApi} from '../BaseApi';

export class UtaBrokerApi extends BaseApi {

    // POST /api/v3/broker/create-sub
    createSub(qsOrBody: object) {
        const url = '/api/v3/broker/create-sub';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/broker/sub-list
    subList(qsOrBody: object) {
        const url = '/api/v3/broker/sub-list';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/broker/modify-sub
    modifySub(qsOrBody: object) {
        const url = '/api/v3/broker/modify-sub';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/broker/sub-withdrawal
    subWithdrawal(qsOrBody: object) {
        const url = '/api/v3/broker/sub-withdrawal';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/broker/sub-deposit-address
    subDepositAddress(qsOrBody: object) {
        const url = '/api/v3/broker/sub-deposit-address';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/broker/all-sub-deposit-withdrawal
    allSubDepositWithdrawal(qsOrBody: object) {
        const url = '/api/v3/broker/all-sub-deposit-withdrawal';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/broker/create-sub-apikey
    createSubApikey(qsOrBody: object) {
        const url = '/api/v3/broker/create-sub-apikey';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/broker/modify-sub-apikey
    modifySubApikey(qsOrBody: object) {
        const url = '/api/v3/broker/modify-sub-apikey';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/broker/delete-sub-apikey
    deleteSubApikey(qsOrBody: object) {
        const url = '/api/v3/broker/delete-sub-apikey';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/broker/query-sub-apikey
    querySubApikey(qsOrBody: object) {
        const url = '/api/v3/broker/query-sub-apikey';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/broker/commission
    commission(qsOrBody: object) {
        const url = '/api/v3/broker/commission';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v3/UtaBrokerApi.java
package com.bitget.openapi.api.v3;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface UtaBrokerApi {

    @POST("/api/v3/broker/create-sub")
    Call<ResponseResult> createSub(@Body Map<String, Object> paramMap);

    @GET("/api/v3/broker/sub-list")
    Call<ResponseResult> subList(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/broker/modify-sub")
    Call<ResponseResult> modifySub(@Body Map<String, Object> paramMap);

    @POST("/api/v3/broker/sub-withdrawal")
    Call<ResponseResult> subWithdrawal(@Body Map<String, Object> paramMap);

    @POST("/api/v3/broker/sub-deposit-address")
    Call<ResponseResult> subDepositAddress(@Body Map<String, Object> paramMap);

    @GET("/api/v3/broker/all-sub-deposit-withdrawal")
    Call<ResponseResult> allSubDepositWithdrawal(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/broker/create-sub-apikey")
    Call<ResponseResult> createSubApikey(@Body Map<String, Object> paramMap);

    @POST("/api/v3/broker/modify-sub-apikey")
    Call<ResponseResult> modifySubApikey(@Body Map<String, Object> paramMap);

    @POST("/api/v3/broker/delete-sub-apikey")
    Call<ResponseResult> deleteSubApikey(@Body Map<String, Object> paramMap);

    @GET("/api/v3/broker/query-sub-apikey")
    Call<ResponseResult> querySubApikey(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/broker/commission")
    Call<ResponseResult> commission(@QueryMap Map<String, String> paramMap);
}
